ibis/expr/operations/arrays.py

```diff
@@ -186,3 +187,14 @@ def dtype(self):
                 }
             )
         )
+
+
+@public
+class ArrayFlatten(Value):
+    # the input must have at least one level of nested for flatten to make sense
+    arg: Annotated[Value[dt.Array], Attrs(dtype=Attrs(value_type=dt.Array))]
```

Comment:
Following should work actually:

```suggestion
    arg:Value[dt.Array[dt.Array]]
```

Reply:
Nice, much better!